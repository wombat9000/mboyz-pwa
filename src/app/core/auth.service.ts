import {Injectable} from '@angular/core';
import {Observable} from 'rxjs/Observable';
import {AngularFireAuth} from 'angularfire2/auth';
import * as firebase from 'firebase';
import {UserFirestore} from './user-firestore.service';
import {Router} from '@angular/router';


export interface User {
  uid: string;
  email: string;
  photoURL?: string;
  displayName?: string;
}

@Injectable()
export class AuthService {

  user: Observable<User>;

  constructor(private afAuth: AngularFireAuth,
              private userRepository: UserFirestore,
              private router: Router) {
    this.user = this.afAuth.authState.switchMap(user => {
      if (user) {
        return userRepository.observeById(user.uid);
      }
      return Observable.of(null);
    });
  }

  isSignedIn(): Observable<boolean> {
    return this.user
      .take(1)
      .map(user => !!user);
  }

  async signOut(): Promise<any> {
    await this.afAuth.auth.signOut();
    return this.router.navigate(['/login']);
  }

  facebookLogin() {
    const provider = new firebase.auth.FacebookAuthProvider();
    return this.oAuthLogin(provider);
  }

  private async oAuthLogin(provider) {
    const credential = await this.afAuth.auth.signInWithPopup(provider);

    return this.updateUserData(credential);
  }

  private updateUserData(credential) {
    const user = credential.user;

    const data: User = {
      uid: user.uid,
      email: user.email,
      displayName: user.displayName,
      photoURL: user.photoURL
    };

    return this.userRepository.save(data);
  }
}
