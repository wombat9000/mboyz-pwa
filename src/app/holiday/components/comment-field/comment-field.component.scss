@import '~@angular/material/theming';
@import '../../../../mboyz-theme';

:host {
  margin-top: 5px;
  background-color: lighten(mat-color($mboyz-primary, 300), 30%);
  padding-top: 4px;
  padding-left: 5px;
  padding-right: 5px;

  mat-form-field {
    width: 100%;
    label.mat-form-field-label {
      font-size: 80%;
    }
  }
}
