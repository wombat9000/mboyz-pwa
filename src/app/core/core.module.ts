import {AppComponent} from './containers/app.component';
import {BgImageDirective} from './components/user-menu/bg-image.directive';
import {NgModule} from '@angular/core';
import {UserMenuComponent} from './components/user-menu/user-menu.component';
import {TopBarComponent} from './containers/top-bar/top-bar.component';
import {RouterModule} from '@angular/router';
import {MaterialModule} from '../material';

export const COMPONENTS = [
  AppComponent,
  TopBarComponent,
  UserMenuComponent,
  BgImageDirective,
];

@NgModule({
  imports: [RouterModule, MaterialModule],
  declarations: COMPONENTS,
  exports: COMPONENTS,
})
export class CoreModule {
  static forRoot() {
    return {
      ngModule: CoreModule,
    };
  }
}
