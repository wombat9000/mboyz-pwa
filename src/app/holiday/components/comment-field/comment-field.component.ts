import {Component, EventEmitter, Output} from '@angular/core';

@Component({
  selector: 'app-comment-field',
  template: `
    <mat-form-field>
    <textarea matInput
              matTextareaAutosize
              [(ngModel)]="value"
              (keyup.enter)="submit()"
              placeholder="Kommentieren ..."></textarea>
    </mat-form-field>
  `,
  styleUrls: ['./comment-field.component.scss']
})
export class CommentFieldComponent {
  value = '';
  @Output()
  submitComment = new EventEmitter<string>();

  submit() {
    this.submitComment.emit(this.value);
    this.value = '';
  }
}
