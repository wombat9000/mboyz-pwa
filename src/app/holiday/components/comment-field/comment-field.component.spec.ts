import {async, ComponentFixture, TestBed} from '@angular/core/testing';
import {CommentFieldComponent} from './comment-field.component';
import {CUSTOM_ELEMENTS_SCHEMA} from '@angular/core';
import {NoopAnimationsModule} from '@angular/platform-browser/animations';
import {FormsModule} from '@angular/forms';
import {By} from '@angular/platform-browser';

describe('CommentFieldComponent', () => {
  let component: CommentFieldComponent;
  let fixture: ComponentFixture<CommentFieldComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [FormsModule, NoopAnimationsModule],
      declarations: [CommentFieldComponent],
      schemas: [CUSTOM_ELEMENTS_SCHEMA]
    })
      .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(CommentFieldComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('produces output when submitting a comment', async (done) => {
    component.submitComment.subscribe(async (it: string) => {
      await fixture.whenStable();
      expect(it).toBe('someComment');
      expect(component.value).toBe('');
      done();
    });

    const input = fixture.debugElement.query(By.css('textarea'));
    input.nativeElement.value = 'someComment';
    input.nativeElement.dispatchEvent(new Event('input'));
    input.nativeElement.dispatchEvent(new KeyboardEvent('keyup', {
      'key': 'Enter'
    }));
  });
});
