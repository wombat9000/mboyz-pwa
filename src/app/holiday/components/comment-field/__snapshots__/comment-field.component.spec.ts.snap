// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentFieldComponent should compile 1`] = `
<app-comment-field
  submitComment={[Function EventEmitter]}
  value=""
>
  <mat-form-field>
    <textarea
      class="ng-untouched ng-pristine ng-valid"
      matinput=""
      mattextareaautosize=""
      ng-reflect-model=""
      placeholder="Kommentieren ..."
    />
  </mat-form-field>
</app-comment-field>
`;
